core/src/main/scala/chisel3/Printf.scala

```diff
@@ -86,13 +86,31 @@ object printf {
   )(sourceInfo: c.Tree
   ): c.Tree = {
     import c.universe._
+
+    val errorString = "The s-interpolator prints the Scala .toString of Data objects rather than the value " +
+      "of the hardware wire during simulation. Use the cf-interpolator instead. If you want " +
+      "an elaboration time print, use println."
+
+    // Find a Data in the AST by matching on the Scala 2.13 string
+    // concatentation pattern
+    def findChiselData(x: c.Tree): Unit = x match {
+      case q"$x+$y" => {
+        if (x.tpe <:< typeOf[chisel3.Data] || y.tpe <:< typeOf[chisel3.Data]) {
+          c.error(c.enclosingPosition, errorString)
+        } else {
+          findChiselData(x)
+          findChiselData(y)
+        }
+      }
+      case _ =>
+    }
+    findChiselData(fmt)
```

Comment:
```suggestion
    throwOnChiselData(fmt)
```

Reply:
I agree with changing the name, otherwise this LGTM